### ISSUE TITLE
Join ways with coincident ends

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2925]
**[Submitted to the original trac issue database at 1.00pm, Sunday, 2nd May 2010]**

When copying features from Shape Files, ways may need to be joined, but the shapefile importer doesn't create a single node for the common point.  So it's necessary to merge coincident nodes before the ways can be joined.  The attached patch automates this, and I can submit it if wanted.

I've been using it for some OS OpenData import, and I'm happy with its performance.

**Comment by tomhughes** [comment]
**Author: Koying**
**[Added to the original trac issue at 2.36pm, Sunday, 2nd May 2010]**

It is a nice functionality and I trust you tested it so please go ahead.

It would be great if the 2 "mergeNodes" could be, er, merged to avoid code duplication but I can handle it if you don't feel like doing it.

**Comment by tomhughes** [comment]
**Author: tms13**
**[Added to the original trac issue at 3.16pm, Sunday, 2nd May 2010]**

It's intentional to have a different mergeNodes, to make the simple case more efficient.  I'll check it in as is, and then look at unifying the two.

Actually, I just realised that I moved it away from the public one, but I should have just declared it up-front and kept the code together.  So I'll change that before checkin.
